aopy/preproc.py

```diff
@@ -258,14 +259,26 @@ def get_measured_frame_timestamps(estimated_timestamps, measured_timestamps, lat
         search_radius (float, optional): how far away to look for a measurement before giving up
 
     Returns:
-        tuple: tuple containing:
-
-            corrected_timestamps (nframes): measured timestamps with missing values filled in with the next non-nan value
-            uncorrected_timestamps (nframes): some of which will be empty if they were not displayed
+            nframes array: measured timestamps, some of which will be np.nan if they were not displayed
     '''
 
     approx_timestamps = estimated_timestamps + latency_estimate
-    uncorrected_timestamps = find_measured_event_times(approx_timestamps, measured_timestamps, search_radius)
+    return find_measured_event_times(approx_timestamps, measured_timestamps, search_radius)
+
+def fill_missing_timestamps(uncorrected_timestamps):
+    '''
+    Fill missing timestamps by copying the subsequent timestamp over any NaNs. For example, if you have
+    timestamps `[0.01, 0.08, np.nan, np.nan, 0.25, np.nan, 0.38]`, then apply fill_missing_timestamps, 
+    the result would be `[0.01, 0.08, 0.25, 0.25, 0.25, 0.38, 0.38]`. Used by proc_exp() to give the times
+    at which things appeared on the screen, since sometimes the screen will miss a refresh period and not 
+    display something until the next cycle.
+
+    Args:
+        uncorrected_timestamps (nframes): some of which can be np.nan
```

Comment:
I think this arg could use a more specific definition. 'uncorrected_timestamps' should include NaNs or else this function doesn't need to be called. Maybe indicate where the uncorrected timestamps will typically come from?